### PR TITLE
feat: add twitter dialog after swap

### PR DESCRIPTION
**Pull request by luizstacio**
Ask users for sharing that they are using SwaySwap on twitter

### CODE REVIEW

File: packages/app/src/systems/Tweet/components/TwitterDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        </p>
+        <Link href={tweetLink} isExternal className="hover:no-underline">
+          <Button
+            size="md"
+            variant="primary"
+            isFull
+            className="mt-6"
+            onPress={shareTweet}
+          >
+            Share on twitter
```
Comment (adlerjohn) [inline packages/app/src/systems/Tweet/components/TwitterDialog.tsx:50]:
```suggestion
            Share on Twitter
```

File: packages/app/src/systems/Tweet/components/TwitterDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+          </Button>
+        </Link>
+        <Button
+          size="sm"
+          isFull
+          className="mt-2 p-0 border-0 text-gray-400"
+          onPress={handleSkip}
+        >
+          I don&apos;t have a Twitter account
```
Comment (adlerjohn) [inline packages/app/src/systems/Tweet/components/TwitterDialog.tsx:59]: Hmm. Given that this is a share rather than e.g. a request for funds, saying you don't have a Twitter account isn't aligned. Maybe something like:

```suggestion
          No thanks
```

?

File: packages/app/src/systems/Tweet/components/TwitterDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import { useEffect } from "react";
+
+import { useTwitterDialog } from "../hooks/useTwitterDialog";
+
+import { Button, Dialog, Link, useDialog } from "~/systems/UI";
+
+const tweetText = `I'm using #SwaySwap, a blazingly fast DEX built on Fuel @fuellabs_\nhttps://fuellabs.github.io/swayswap`;
```
Comment (adlerjohn) [inline packages/app/src/systems/Tweet/components/TwitterDialog.tsx:7]: Can this be replaced with some placeholder random text for now? You can just comment this line so it'll be east to change later.

Comment (luizstacio) [reply packages/app/src/systems/Tweet/components/TwitterDialog.tsx:7]: Remove the entire text?

Comment (adlerjohn) [reply packages/app/src/systems/Tweet/components/TwitterDialog.tsx:7]:
```suggestion
// const tweetText = `I'm using #SwaySwap, a blazingly fast DEX built on Fuel @fuellabs_\nhttps://fuellabs.github.io/swayswap`;
const tweetText = `foo`;
```

File: packages/app/src/styles/components/welcome-page.css [hunk 4]
```diff
@@ -91,6 +91,10 @@
     }
     img {
       @apply mx-auto mb-5;
+
+      @media (max-width: 640px) {
+        @apply h-[180px];
+      }
```
Comment (adlerjohn) [inline packages/app/src/styles/components/welcome-page.css:97]: Can you do this fix in a separate PR? That way it can be reviewed atomically.

**Comment by luizstacio** [comment]
@adlerjohn maybe wait to the tweet text to merge it?

**Comment by adlerjohn** [comment]
That's also a possibility, we could hold off on merging this until we're ready. That would mean this branch would need to be kept up to date, but it won't be for too long.

**Comment by pedronauck** [comment]
Screen Shot 2022-06-20 at 16 46 39

it's a lit bit awry layout here

File: packages/app/src/systems/Tweet/components/TwitterDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
+import { useEffect } from "react";
+
+import { useTwitterDialog } from "../hooks/useTwitterDialog";
+
+import { Button, Dialog, Link, useDialog } from "~/systems/UI";
+
+// const tweetText = `I'm using #SwaySwap, a blazingly fast DEX built on Fuel @fuellabs_\nhttps://fuellabs.github.io/swayswap`;
```
Comment (pedronauck) [inline packages/app/src/systems/Tweet/components/TwitterDialog.tsx:7]: this will remain with this fake version?

**Comment by adlerjohn** [comment]
Is that related to the current PR @pedronauck?

File: packages/app/src/systems/Tweet/hooks/useTwitterDialog.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { atom, useAtom } from 'jotai';
+
+const TWITTER_DIALOG_KEY = 'swayswap-dialog';
```
Comment (pedronauck) [inline packages/app/src/systems/Tweet/hooks/useTwitterDialog.ts:3]: why do we need to persist at localstorage here?

**Review by pedronauck — CHANGES_REQUESTED** [review]

File: packages/app/src/systems/Tweet/hooks/useTwitterDialog.ts (hunk 6 above)
Comment (luizstacio) [reply packages/app/src/systems/Tweet/hooks/useTwitterDialog.ts:3]: The popup should be open only once, after the user takes an action, share or dismiss we never show again.

File: packages/app/src/systems/Tweet/components/TwitterDialog.tsx (hunk 5 above)
Comment (luizstacio) [reply packages/app/src/systems/Tweet/components/TwitterDialog.tsx:7]: We are going to hold until we have a final text

Comment (adlerjohn) [reply packages/app/src/systems/Tweet/components/TwitterDialog.tsx:7]: The text is final. To clarify, my comment about weren't about the finality of the text, but rather to not have a running version of SwaySwap with the above text that someone might Tweet by accident before we're ready.

**Review by adlerjohn — DISMISSED** [review]

**Review by LuizAsFight — CHANGES_REQUESTED** [review]
`swayswap-dialog` is too generic for a speciic twitter dialog.

I think it could be something like: `swayswap-share-twitter-dialog`

**Review by LuizAsFight — APPROVED** [review]
LGTM

**Review by pedronauck — APPROVED** [review]
